File: lib/starvation.ex
defmodule Starvation do

  def count(max, start \\ now) when is_integer(max) do
    spawn fn ->
      receive do
        i when max >= i ->
          IO.puts "#{inspect self} Received #{i}"
          actor = count(max, start)
          send(actor, i + 1)
          IO.puts "#{inspect self} Sleeping for #{max} ms"
          :timer.sleep(max)
          IO.puts "#{inspect self} Actor exiting..."
        i ->
          done = now
          IO.puts "Done at #{i} in just #{(done - start)} s"
      end
    end
  end

  def now, do: :calendar.datetime_to_gregorian_seconds(:calendar.universal_time)

end

File: Starvation.scala
import scala.actors.Actor
import scala.actors.Actor._

class Starvation(max: Int, start: Long) extends Actor {
  def act() = {
    react {
      case i : Int if max >= i =>
        print(Thread.currentThread + ": Received " + i)
        val s = new Starvation(max, start)
        s.start
        s ! i + 1
        println(" ... starting new actor and sleeping for " + max + "ms")
        Thread.sleep(max)
        println(Thread.currentThread + " actor exiting...")
      case i : Int =>
        println(Thread.currentThread + ": Done in " + (System.currentTimeMillis() - start) + "ms")
    }
  }
}

val s = new Starvation(1000000, System.currentTimeMillis())
s.start
s ! 0
